unixfs/io/dagreader.go

```diff
@@ -19,14 +20,29 @@ var ErrIsDir = errors.New("this dag node is a directory")
 type DagReader struct {
 	serv         mdag.DAGService
 	node         *mdag.Node
-	buf          io.Reader
-	fetchChan    <-chan *mdag.Node
+	pbdata       *ftpb.Data
+	buf          ReadSeekCloser
+	promises     []mdag.NodeGetter
 	linkPosition int
+	offset       int64
```

Comment:
document. i would not feel comfortable jumping in and modifying. a quick comment next to each field, and a more thorough comment for the whole object describing how it operates, would be very helpful.